### DISCUSSION TITLE
Failed to create TUN device in using windows-libipsec branch

### DISCUSSION THREAD

**Discussion by xtsheng123** [strongswan/strongswan#766] [Q&A]
1. On windows platform, does not support the installation of virtual IP addresses #727.  So I use windows-libipsec branch  to avoid this problem   (https://github.com/strongswan/strongswan/discussions/727)
2. The strongswan configuration is compiled as follows:
`CFLAGS="-g -O2 -Wall -Wno-pointer-sign -Wno-format-security \
        -Wno-format -mno-ms-bitfields \
        -I/root/project/strongswan/install_windows_libipsec/include" \
LDFLAGS="-L/root/project/strongswan/install_windows_libipsec/lib" \
./configure --disable-defaults --enable-monolithic --enable-static --enable-svc \
--enable-ikev2 --enable-aes --enable-des --enable-sha1 --enable-sha2 --enable-md5 --enable-ikev1 --enable-openssl --enable-nonce --enable-pem --enable-pkcs1 \
--enable-x509 --enable-socket-win --enable-kernel-wfp --enable-kernel-iph --enable-pubkey --enable-xauth-generic --enable-kernel-libipsec --enable-libipsec \
--enable-eap-identity --enable-eap-mschapv2 --enable-pki --enable-swanctl --with-swanctldir=swanctl --with-strongswan-conf=strongswan.conf \
--host=x86_64-w64-mingw32 --prefix="/root/project/strongswan/install_windows_libipsec"`

3. The error is as follows：
G:\windows_install_file\install_file\bin>charon-svc.exe
00[DMN] Starting IKE service charon-svc (strongSwan 5.4.1dr1, Windows Client 6.1.7601 (SP 1.0)
00[KNL] failed to create TUN device
00[LIB] plugin 'kernel-libipsec': failed to load - kernel_libipsec_plugin_create returned NULL
00[LIB] loaded plugins: charon-svc aes des sha2 sha1 md5 nonce x509 pubkey pkcs1 pem openssl kernel-wfp kernel-iph socket-win vici eap-identity eap-mschapv2 xauth-generic
00[JOB] spawning 16 worker threads

4. I hope you can help solve this problem

**Comment by tobiasbrunner** [comment]
There is currently no [`tun_device_t` implementation](https://github.com/strongswan/strongswan/blob/master/src/libstrongswan/networking/tun_device.c) for Windows.
